statsd/statsd.go

```diff
@@ -888,13 +881,11 @@ func (c *Client) Close() error {
 
 // isOriginDetectionEnabled returns whether the clients should fill the container field.
 //
-// If DD_ENTITY_ID is set, we don't send the container ID
-// If a user-defined container ID is provided, we don't ignore origin detection
-// as dd.internal.entity_id is prioritized over the container field for backward compatibility.
-// If DD_ENTITY_ID is not set, we try to fill the container field automatically unless
-// DD_ORIGIN_DETECTION_ENABLED is explicitly set to false.
-func isOriginDetectionEnabled(o *Options, hasEntityID bool) bool {
-	if !o.originDetection || hasEntityID || o.containerID != "" {
+// Disable origin detection only in one of the following cases:
+// - DD_ORIGIN_DETECTION_ENABLED is explicitly set to false
+// - o.originDetection is explicitly set to false, which is true by default
+func isOriginDetectionEnabled(o *Options) bool {
+	if !o.originDetection || o.containerID != "" {
 		// originDetection is explicitly disabled
 		// or DD_ENTITY_ID was found
```

Comment:
This comment should also be updated.

Reply:
Also here: https://github.com/DataDog/datadog-go/blob/master/statsd/options.go#L378